src/AnonymousUtils.js

```diff
@@ -0,0 +1,113 @@
+/**
+ * Copyright (c) 2015-present, Parse, LLC.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @flow-weak
+ */
+import ParseUser from './ParseUser';
+const uuidv4 = require('uuid/v4');
+
+let registered = false;
+
+const authenticationProvider = {
+  authData: {
```

Comment:
can you name the top level 'provider'? little confused as to why we would have authData: authData: